engine/src/main/java/org/hibernate/validator/internal/cfg/context/PropertyConstraintMappingContextImpl.java

```diff
@@ -64,17 +57,17 @@ protected PropertyConstraintMappingContextImpl getThis() {
 
 	@Override
 	public PropertyConstraintMappingContext constraint(ConstraintDef<?, ?> definition) {
-		if ( member instanceof Field ) {
+		if ( property instanceof JavaBeanField ) {
```

Comment:
Our abstraction doesn't work very well here.

That's why I'm not a big fan of merging field and getters so early and not being able to have a distinction between them at the base layer (by base, I mean `Property`, `Callable` and so on).

TBH, considering the behavior is entirely different, I'm not a big fan of this class. Note that the issue was preexisting, it's not something you introduced.